app/delegators/delegator_decorator.rb

```diff
@@ -0,0 +1,17 @@
+module DelegatorDecorator
+  def initialize(obj)
+    super obj
+    @delegate_sd_obj = obj
+  end
+
+  def __getobj__
+    @delegate_sd_obj
+  end
+  alias_method :model, :__getobj__
+
+  def __setobj__(obj)
+    @delegate_sd_obj = obj
+  end
+end
+
+Delegator.prepend DelegatorDecorator
```

Comment:
This pings my naming 'spidey sense'. Should we be more descriptive with our Decorator names as to define, top of file what exactly they are doing (aside from the decoration process)